src/Tomlyn/Parsing/Lexer.cs

```diff
@@ -525,7 +525,7 @@ private void ReadNumberOrDate(char32? signPrefix = null, TextPosition? signPrefi
                     previousCharIsDigit = _c == '0';
                     if (previousCharIsDigit)
                     {
-                        _textBuilder.Append(_c);
+                        _textBuilder.Append(_c.ToString());
```

Comment:
```suggestion
                        _textBuilder.Append((char)_c);
```

> Because _c is a char32 

Oh right, forgot without looking at the code.